modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/rebalance/DataGenerationApplication.java

```diff
@@ -0,0 +1,120 @@
+import org.apache.ignite.internal.ducktest.utils.IgniteAwareApplication;
+
+/**
+ *
+ */
+public class DataGenerationApplication extends IgniteAwareApplication {
+    /** Max streamer data size. */
+    private static final long MAX_STREAMER_DATA_SIZE = 100_000_000;
+
+    /** {@inheritDoc} */
+    @Override protected void run(JsonNode jsonNode) throws Exception {
+        int backups = jsonNode.get("backups").asInt();
+        int cacheCnt = jsonNode.get("cacheCount").asInt();
+        int entryCnt = jsonNode.get("entryCount").asInt();
+        int entrySize = jsonNode.get("entrySize").asInt();
+
+        log.info("Data generation started [backups=" + backups + ", cacheCount=" + cacheCnt
+            + ", entryCount=" + entryCnt + ", entrySize=" + entrySize + "]");
+
+        for (int i = 1; i <= cacheCnt; i++) {
+            IgniteCache<Integer, DataModel> cache = ignite.createCache(
+                new CacheConfiguration<Integer, DataModel>("test-cache-" + i)
+                    .setBackups(backups));
+
+            generateCacheData(cache.getName(), entryCnt, entrySize);
+        }
+
+        log.info("Data generation finished.");
+
+        markSyncExecutionComplete();
+    }
+
+    /**
+     * @param cacheName Cache name.
+     * @param entryCnt Entry count.
+     * @param entrySize Entry size.
+     */
+    private void generateCacheData(String cacheName, int entryCnt, int entrySize) {
+        int logStreamedEntriesQuant = (int)Math.pow(10, (int)Math.log10(entryCnt) - 1);
+
+        try (IgniteDataStreamer<Integer, DataModel> stmr = ignite.dataStreamer(cacheName)) {
+            for (int i = 0, n = 0; i < entryCnt; i++) {
+                stmr.addData(i, new DataModel(entrySize));
+
+                int streamed = i + 1;
+
+                if (streamed % logStreamedEntriesQuant == 0)
+                    log.info("Streamed " + streamed + " entries into " + cacheName);
+
+                n += entrySize;
+
+                if (n >= MAX_STREAMER_DATA_SIZE) {
+                    n = 0;
+
+                    stmr.flush();
+                }
+            }
+        }
+
+        if (entryCnt % logStreamedEntriesQuant != 0)
+            log.info("Streamed " + entryCnt + " entries into " + cacheName);
+
+        log.info(cacheName + " data generated.");
+    }
+
+    /**
+     *
+     */
+    private static class DataModel {
+        /** Cached payload. */
+        private static byte[] cachedPayload;
+
+        /** Payload. */
+        final byte[] payload;
+
+        /**
+         * @param entrySize Entry size.
+         */
+        DataModel(int entrySize) {
+            payload = getPayload(entrySize);
+        }
+
+        /**
+         * @param payloadSize Payload size.
+         */
+        private static byte[] getPayload(int payloadSize) {
+            if (cachedPayload == null || cachedPayload.length != payloadSize) {
+                cachedPayload = new byte[payloadSize];
+
+                new Random(42).nextBytes(cachedPayload);
```

Comment:
Please, use `ThreadLocalRandom` here.

Reply:
Using Random instance with constant seed guarantees that data will be same for all runs.